### ISSUE TITLE
"editorconfig" configuration prints an "Ignored unknown option" warning

### ISSUE THREAD

**Issue by Shayan-To** [prettier/prettier#13539]
Adding `{ "editorconfig": true }` to the configuration file, although it works, but prints `[warn] Ignored unknown option { editorconfig: true }.` (multiple times).

**Environments:**

-    Prettier Version: 2.7.1
-    Running Prettier via: CLI
-    Runtime: Node.js v16.14.2
-    Operating System: Ubuntu 20.04.4
-    Prettier plugins (if any): --

**Steps to reproduce:**
Add `{ "editorconfig": true }` to `.prettierrc.json` and run prettier.

**Expected behavior:**
No warn message is printed.

**Actual behavior:**
This message is printed (multiple times): `[warn] Ignored unknown option { editorconfig: true }.`

**Comment by thorn0** [comment]
That's an option of Prettier CLI, not something specified in a config file.

**Comment by Shayan-To** [comment]
@thorn0 So I think it should be removed from the [json schema](http://json.schemastore.org/prettierrc) (it is the fifth property).

**Comment by thorn0** [comment]
Indeed, it shouldn't be there unless I'm missing something. Need to investigate. Thanks for noticing this.

**Comment by danielpza** [comment]
> That's an option of Prettier CLI, not something specified in a config file.

Is there any reason it cannot be specified in the config file? It actually recognizes the option if placed there. 

The docs makes no mention of this either https://prettier.io/docs/en/configuration.html#editorconfig

**Comment by kotx** [comment]
Why isn't this an option on .prettierrc?

**Comment by bschaeublin** [comment]
This problem has been bothering me for some time. If I set `editorconfig: true` as specified in the config scheme, the editorconfig settings are respected when saved in webstorm (with prettier enabled).

But the ci check `npx prettier --check .` gives a warning about this "unknown" option.
It would be nice if this option was still supported in the prettier-options file, so that IDE integrations don't have to be configured separately.

**Comment by danielpza** [comment]
Can a prettier maintainer please address this issue? It doesn't make sense to be a cli option but not in the config file. Is the `editorconfig` option going to be supported in the .prettierrc file or not?

**Comment by rotu** [comment]
I'm finding that "editorconfig" is *not* respected in `.prettierrc.json`. It's true by default, and if I set to false in the config file, `prettier` still pulls settings from `.editorconfig`.
